### PR TITLE
docs: Update designs with MetricsLogger

### LINKED ISSUE #940: Sentry - Update designs with MetricsLogger
**Description**

**Acceptance Criteria**
- [ ] Update designs to include Metrics Logger, so that implementation could be easily swapped or extended (add crashalitycs or completely replace Sentry with Crashalytics)

### PR DESCRIPTION
**Pull request by solid-vladlaba**
Fixes #940

## Checklist

- [x] Documented

### CODE REVIEW

**Review by illia-romanenko — CHANGES_REQUESTED** [review]
Great job @solid-vladlaba @solid-dimakoniaiev! Excited to see all the design changes and they look awesome. I've added a couple of suggestions - please let me know what you think. 
Overall this is a very solid document that I enjoyed reading, diagrams very helpful!

File: metrics/web/docs/features/metrics_logger/diagrams/metrics_logger_class_diagram.puml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  class LoggerWriter {
+    + writeError(Object error, [Object stackTrace]) : FutureOr<void>
+    + setContext(String key, Map<String, dynamic> context) : FutureOr<void>
+  }
+
+  class MetricsLogger {
+    - _loggerWriter : LoggerWriter
+    + static initialize(LoggerWriter loggerWriter,  Map<String, dynamic> contexts) : Future<void>
+    + static logError(Object error, [Object stackTrace]) : Future<void>
```
Comment (illia-romanenko) [inline metrics/web/docs/features/metrics_logger/diagrams/metrics_logger_class_diagram.puml:14]: Can we specify stackTrace type to be StackTrace? Can you ping me so we can discuss this - I feel that I miss something here.

**Review by illia-romanenko — APPROVED** [review]
Phenomenal work @solid-vladlaba & @solid-dimakoniaiev - this document is a great exploration of the Logger approach, Sentry, and ready to be extended with any other logging library.

**Review by solid-vovabeloded — APPROVED** [review]
LGTM, thanks @solid-vladlaba & @solid-dimakoniaiev! The document looks awesome and clearly explains the implementation of the `MetricsLogger`!

**Pull request merged**